extensions/amp-story-subscriptions/0.1/amp-story-subscriptions.js

```diff
@@ -1,31 +1,84 @@
+import {iterateCursor} from '#core/dom';
 import * as Preact from '#core/dom/jsx';
 import {Layout_Enum} from '#core/dom/layout';
 
 import {CSS} from '../../../build/amp-story-subscriptions-0.1.css';
+import {
+  StateProperty,
+  getStoreService,
+} from '../../amp-story/1.0/amp-story-store-service';
 
 const TAG = 'amp-story-subscriptions';
 
+/**
+ * The index of the limited-content page, which is the page where the paywall would be triggered.
+ * @const {number}
+ */
+const FIRST_PAYWALL_STORY_PAGE_INDEX = 2;
+
 export class AmpStorySubscriptions extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
+
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = getStoreService(this.win);
```

Comment:
Done. But why is it reducing the bundle size? My original thought is that their only differences are just one is async and the other one is sync? How is that going to impact bundle size? Or is it just because getStoreService from amp-story-store-service is much heavier than Services.storyStoreServiceForOrNull from #services?

Reply:
The `getStoreService` method needs to bundle the whole service because it contains the line `new AmpStoryStoreService`. This is why it can do it sync, because if there's no service it will create it.

The async version doesn't have the capability to create the service because it doesn't bundle it into the extension, which is what we want here

Reply:
It has to do with the way the compilation works. If you think of an extension as a unique piece of JS, anything that the bundle depends on, it will be included on the JS bundle. `getStoreService` is written in a way that it requires the code for the store service (because it's possible that it will instantiate the service), but `Services.storeServiceForNull` doesn't (it just grabs a value from a dictionary). You can look into the source code of each function to understand why

Reply:
Yeah that makes sense. Just had an offline sync with Caleb too. Wasn't familiar with how AMP core runtime's global variable for services work. Also forgot about the extensions are imported individually so no way to access those other extensions if not included in the bundle. Thanks for the info both of you!